Lint paths per Coursera frontend file organization

turadg opened the pull request:
ptal @jnwng @eleith

eleith commented on src/lint/linter/ArcanistCourserorgLinter.php:
```diff
@@ -0,0 +1,87 @@
+  public function getInfoDescription() {
+    return pht(
+      'Stifles developer creativity by requiring files to conform to '.
+      'Coursera Frontend File Organization, by edict of the FCC.');
```
is their a creativity stifling document we could link to?

turadg replied:
I'll draft a memo and get consensus to publish

eleith commented on src/lint/linter/ArcanistCourserorgLinter.php:
```diff
@@ -0,0 +1,87 @@
+            'Every bundle with components must have '.
+            'an NLS file.'));
+
+    } else if (preg_match('/^static\/(.+)\/components/', $path)) {
```
don't we have components in open-course? or have all those moved out?

turadg replied:
we still do, yeah, but I'd like those to begin counting as linting errors. they won't block anyone, just show up highlighted in blue in the revision. people can use judgment for when to Request they move.

eleith replied:
@jnwng are you ok with that?

jnwng replied:
i wasn't aware that this was a thing, what happens when we have page-specific utilizations of components? i can see there being things (some that i'm working on now) that just wrap common components but probably not ought to be shared

turadg replied:
The WEB repo schema proposal moves all components out of pages to bundles. The current JSX localization pipeline depends on it.

jnwng replied:
does this mean that all our open-course components cannot currently be localized? this is a blocker

jnwng commented on src/lint/linter/ArcanistCourserorgLinter.php:
```diff
@@ -0,0 +1,87 @@
+  public function lintPath($path) {
+    if (preg_match('/([^_]*)__styles__\/(.+).styl/', $path, $matches)) {
+      $jsxFilename = $matches[1].$matches[2].'.jsx';
+      print_r($jsxFilename);
```
remove print?

jnwng commented on src/lint/linter/ArcanistCourserorgLinter.php:
```diff
@@ -0,0 +1,87 @@
+        $this->raiseLintAtPath(
+          self::MISPLACED_COMPONENT_STYL,
+          pht(
+            'Stylus in __styles__ must correspond to a JSX in parent components.'));
```
this is slightly strange wording, perhaps "Stylus files in __styles__ must have a corresponding React component"?

turadg commented:
@jnwng I cleaned up the print and help string. I also added a comment on the components in bundles idea that we've been discussing in person.

good to land?

jnwng commented:
:+1:
